packages/flame/lib/src/sprite_sheet.dart

```diff
@@ -27,30 +27,46 @@ class SpriteSheet {
   /// size. If it's an animation sheet, this would be the frame size.
   final Vector2 srcSize;
 
+  /// The empty space around the edges of the image.
+  final double margin;
+
+  /// This empty space in between adjacent tiles within the image.
+  final double spacing;
+
+  final int _rows;
+  final int _columns;
   final Map<int, Sprite> _spriteCache = {};
 
   /// Creates a sprite sheet given the image and the tile size.
   SpriteSheet({
     required this.image,
     required this.srcSize,
-  });
+    this.margin = 0,
+    this.spacing = 0,
+  })  : _columns =
+            (image.width - 2 * margin + spacing) ~/ (srcSize.x + spacing),
+        _rows = (image.height - 2 * margin + spacing) ~/ (srcSize.y + spacing);
 
   SpriteSheet.fromColumnsAndRows({
     required this.image,
     required int columns,
     required int rows,
-  }) : srcSize = Vector2(
-          image.width / columns,
-          image.height / rows,
+    this.spacing = 0,
+    this.margin = 0,
+  })  : _columns = columns,
+        _rows = rows,
+        srcSize = Vector2(
+          (image.width - 2 * margin - (columns - 1) * spacing) / columns,
+          (image.height - 2 * margin - (rows - 1) * spacing) / rows,
         );
 
   /// Compute the number of columns the image has
   /// by using the image width and tile size.
-  int get columns => image.width ~/ srcSize.x;
+  int get columns => _columns;
```

Comment:
Since the columns and rows are final, can't we just make those variable public instead of having a getter?

Reply:
Done